io/src/test/java/io/smallrye/common/io/jar/JarFilesTest.java

```diff
@@ -0,0 +1,19 @@
+package io.smallrye.common.io.jar;
+
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.jar.JarFile;
+
+import org.junit.Test;
+
+public class JarFilesTest {
+
+    @Test
+    public void shouldReadMultiReleaseJars() throws IOException {
+        File tmpFile = MultiReleaseJarGenerator.generateMultiReleaseJar();
+        JarFile jarFile = JarFiles.create(tmpFile);
+        assertTrue(jarFile.isMultiRelease());
```

Comment:
Does this rely on being run on Java 9+? (Not sure if that would be bad or not.)

Reply:
Yes, I should probably enable it for JDK9+ builds too. Thanks!

Reply:
I think it would, which will cause the Java 8 tests to fail.  It might be worth skipping associated tests in that case.

Reply:
Here's an idea: add a JDK-specific `isMultiRelease(JarFile)` method.  You could even make Java 8 recognize them by browsing the manifest.